src/xpcc/architecture/platform/driver/core/cortex/syscalls.cpp.in

```diff
@@ -140,25 +201,47 @@ _isatty(int /*file*/)
  *         Since the _reent struct is not used in the current implementation,
  *         the following messages must be suppressed.
  */
-%% if parameters.allocator == "newlib"
-// Using the NewLib Malloc Implementation
-uint8_t *__brkval = &__heap_start;		// Points to current end of the heap
-extern "C"
 void *
 _sbrk_r(struct _reent *,  ptrdiff_t size)
 {
 	// move heap pointer
 	uint8_t *heap = __brkval;
 	__brkval += size;
 
-	if (__brkval >= &__heap_end) {
+	if (__brkval >= heap_end) {
 		// ERROR: heap and stack collision!
 		abort();
 	}
 
 	//  Return pointer to start of new heap area.
 	return heap;
 }
+
+// FIXME: "Unwrapping" the malloc for newlib
```

Comment:
This could potentially be made nicer, by not setting the linker flags.
But that would require the linker to know about the parameters, which is not a mechanism that we have at the moment.